catroid/src/androidTestPhiro/java/org/catrobat/catroid/uiespresso/ui/activity/PhiroScriptActivityTest.java

```diff
@@ -0,0 +1,119 @@
+import org.catrobat.catroid.content.Project;
+import org.catrobat.catroid.content.Script;
+import org.catrobat.catroid.content.Sprite;
+import org.catrobat.catroid.content.StartScript;
+import org.catrobat.catroid.content.bricks.SetXBrick;
+import org.catrobat.catroid.ui.ScriptActivity;
+import org.catrobat.catroid.ui.fragment.BrickCategoryFragment;
+import org.catrobat.catroid.ui.fragment.FormulaEditorCategoryListFragment;
+import org.catrobat.catroid.uiespresso.util.BaseActivityInstrumentationRule;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+import static android.support.test.espresso.Espresso.onView;
+import static android.support.test.espresso.action.ViewActions.click;
+import static android.support.test.espresso.assertion.ViewAssertions.matches;
+import static android.support.test.espresso.matcher.ViewMatchers.withId;
+
+import static org.catrobat.catroid.ui.SettingsActivity.SETTINGS_SHOW_HINTS;
+import static org.catrobat.catroid.ui.SettingsActivity.SETTINGS_SHOW_PHIRO_BRICKS;
+import static org.catrobat.catroid.uiespresso.util.matchers.AdapterViewMatchers.withAdaptedDataAt;
+import static org.catrobat.catroid.uiespresso.util.matchers.AdapterViewMatchers.withIndex;
+
+@RunWith(AndroidJUnit4.class)
+public class PhiroScriptActivityTest {
+
+	@Rule
+	public BaseActivityInstrumentationRule<ScriptActivity> baseActivityTestRule = new
+			BaseActivityInstrumentationRule<>(ScriptActivity.class, true, false);
+
+	@Before
+	public void setUp() throws Exception {
+		createProject("phiroTestProject");
+		preparePhiroSettings();
+		baseActivityTestRule.launchActivity(null);
+	}
+
+	@Test
+	public void phiroIsFirstInBrickCategoryTest() {
+		onView(withId(R.id.button_add))
+				.perform(click());
+
+		onView(withIndex(withId(android.R.id.list), 0))
+				.check(matches(withAdaptedDataAt("Phiro", 0)));
+	}
+
+	@Test
+	public void phiroIsFirstInFormulaEditorDeviceCategoryTest() {
+		onView(withId(R.id.brick_set_x_edit_text))
+				.perform(click());
+
+		onView(withId(R.id.formula_editor_keyboard_sensors))
+				.perform(click());
+
+		onView(withIndex(withId(android.R.id.list), 1))
+				.check(matches(withAdaptedDataAt("phiro_front_left_sensor", 0)));
+	}
+
+	public static Project createProject(String projectName) {
+		Project project = new Project(null, projectName);
+		Sprite sprite = new Sprite("testSprite");
+		Script script = new StartScript();
+
+		SetXBrick setXBrick = new SetXBrick();
+
+		script.addBrick(setXBrick);
+		sprite.addScript(script);
+		project.getDefaultScene().addSprite(sprite);
+
+		ProjectManager.getInstance().setProject(project);
+		ProjectManager.getInstance().setCurrentSprite(sprite);
+
+		return project;
+	}
+
+	private void preparePhiroSettings() {
+		SharedPreferences sharedPreferences = PreferenceManager.getDefaultSharedPreferences(InstrumentationRegistry.getTargetContext());
+
+		SharedPreferences.Editor editor = sharedPreferences.edit();
```

Comment:
when changing shared preferences in a test you should reset them to what they were before the testrun in a teardown method.